transport/tchannel/handler.go

```diff
@@ -72,18 +70,16 @@ type inboundCallResponse interface {
 	SetApplicationError() error
 }
 
-// responseWriter provides an interface similar to handlerWriter.
-//
-// It allows us to control handlerWriter during testing.
+// responseWriter enhances transport.ResponseWriter interface with transport specific
+// methods.
 type responseWriter interface {
-	AddHeaders(h transport.Headers)
-	AddHeader(key string, value string)
+	transport.ResponseWriter
+
+	AddSystemHeader(key string, value string)
```

Comment:
optional: do you want to use `AddTransportHeader` instead, as usually we differentiate between transport header and application header. System sounds quite general.

Reply:
AddSystemHeader is used in grpc, so I decided to make it consistent at first step (: Maybe we'll return to this and change to AddTransportHeader in a future refactoring, as it makes sense to me.